tests/graph/test_utils.py

```diff
@@ -12,3 +14,24 @@ def test_stack_trace():
         v = vector()
         assert len(v.tag.trace) == 1
         assert len(v.tag.trace[0]) == 2
+
+
+def test_replacements():
```

Comment:
Also I feel we need a couple more test cases, including not replacing a root variable (which `x` is)

```suggestion
def graph_replace():
```